src/paillier.rs

```diff
@@ -80,6 +80,21 @@ impl PaillierEncryptionKey {
         let c = a.modmul(&b, self.0.nn());
         Ok((PaillierCiphertext(c), PaillierNonce(nonce)))
     }
+
+    /// Masks a [`PaillierNonce`] `nonce` with another [`PaillierNonce`] `mask` and exponent `e`
+    /// for use in proving properties of `nonce`'s corresponding ciphertext. Both `nonce` and
+    /// `mask` MUST have been generated by the given [`PaillierEncryptionKey`].
+    ///
+    /// The resulting [`MaskedNonce`] is computed as `nonce * mask^e mod N`, where `N`
+    /// is the modulus of [`PaillierEncryptionKey`].
+    pub(crate) fn mask(
+        &self,
+        nonce: &PaillierNonce,
+        mask: &PaillierNonce,
```

Comment:
I think these are mixed up now. The formula is `mask *  nonce ^ e`. 

In all the places you call this, you pass `mask(r, rho)`, where `r` is the mask and `rho` is the original nonce. So if you rename these two parameters (and fix the corresponding docs), it should be correct.

Reply:
Good catch! I actually prefer the `nonce`, `mask` order, because from a caller perspective it makes sense to input the `nonce` first then all the masking arguments (at least it makes more sense to me :-) ). So I instead updated the callers to use the right order (and fixed the implementation as well). Let me know if you prefer the `mask`, `nonce` order though!